src/os/bluestore/BlueStore.h

```diff
@@ -328,25 +340,25 @@ class BlueStore : public ObjectStore,
       return nullptr;
     }
 
-    Blob *new_blob(Cache *c) {
+    BlobRef new_blob(Cache *c) {
       int64_t id = get_new_id();
       Blob *b = new Blob(id, c);
+      b->get();
```

Comment:
@liewegas. Additional ref_count increment is needed since Blobs are stored in intrusive::set<Blob> blob_map. Blob has to persist while it's stored in this blob_map even if no corresponding BlobRefs are present.
Other options are either using std::set<BlobRef> or intrusive::set<BlobRef> but IMHO they both have some performance/memory footprint drawbacks. Have I missed something and there is better approach?